### PR TITLE
Implement controller logic for v2

### PR DESCRIPTION
**Pull request by peter-madigan**
This defines a new configuration format for the v2 networks. Namely, there is a new configuration file for controllers, controllers now have a miso_us, miso_ds, and mosi network attribute (stored by io group and io channel via controller.networks), and controllers have a new method for initializing hydra networks. 

I've added a script for generated network configuration files for a given asic layout (since these can be a bit tricky to do by hand).

Updates the docs and README for the new v2 interface.

Adds a configuration type field (and verification) for all configuration files.

### CODE REVIEW

File: larpix/io/fakeio.py [hunk 1]
```diff
@@ -85,6 +86,7 @@ def send(self, packets):
         '''
         for packet in packets:
             print(packet)
+        self.sent += [packets]
```
Comment (samkohn) [inline larpix/io/fakeio.py]: I prefer ``self.sent.append(packets)`` but it's up to you

File: test/test_controller.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+def test_controller_network(network_controller):
+    c = network_controller
+    networks = ('miso_us','miso_ds','mosi')
+    for chip_key in c.chips:
+        for network in networks:
+            assert chip_key.chip_id in c.network[chip_key.io_group][chip_key.io_channel][network].nodes()
+    us_links = [(2,3),(2,12),(12,13)]
+    ds_links = [(3,2),(12,2),(13,12),(2,'ext')]
+    mosi_links = [link[::-1] for link in us_links] + [link[::-1] for link in ds_links]
+    assert all([link in c.network[chip_key.io_group][chip_key.io_channel]['miso_us'].edges() for link in us_links])
```
Comment (samkohn) [inline test/test_controller.py]: It looks like this is comparing ``...edges()`` and ``us_links`` to see if they have the same elements. The cleanest way I've seen to do this is ``assert set(...edges()) == set(us_links)``.

File: test/test_controller.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    c.add_network_link('miso_ds',1,1,(3,2),2)
+    assert len(c.network[1][1]['miso_ds'].edges()) == 4
+    assert c.network[1][1]['miso_ds'].edges[(3,2)]['uart'] == 2
+    c.add_network_link('mosi',1,1,(3,2),0)
+    c.add_network_link('mosi',1,1,(2,3),2)
+    assert len(c.network[1][1]['mosi'].edges()) == 7
+    assert c.network[1][1]['mosi'].edges[(3,2)]['uart'] == 0
+    assert c.network[1][1]['mosi'].edges[(2,3)]['uart'] == 2
+
+def test_controller_init(network_controller):
```
Comment (samkohn) [inline test/test_controller.py]: Add another test to see if ``init_network(1, 1)`` works

File: test/test_controller.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (samkohn) [inline test/test_controller.py]: Don't need these extra lines

File: larpix/controller.py [hunk 5]
```diff
@@ -35,122 +37,337 @@ class Controller(object):
...
+        :param chip_key: chip key to specify unique chip
+
+        '''
+        chip_key = Key(chip_key)
+        io_channel, io_group = chip_key.io_channel, chip_key.io_group
+        del self.chips[chip_key]
+
+        self.network[io_group][io_channel]['mosi'].remove_node(chip_key.chip_id)
+        self.network[io_group][io_channel]['miso_us'].remove_node(chip_key.chip_id)
+        self.network[io_group][io_channel]['miso_ds'].remove_node(chip_key.chip_id)
```
[the remark this answers is not in the document]
Comment (peter-madigan) [reply larpix/controller.py]: It's automatic

**Pull request merged**